### PR TITLE
Added tutorial on macronizing (based on example file)

### PR DESCRIPTION
**Pull request by soosub**
## Context for changes

Hey @sduquemesa , I was working through the examples folder with jupy notebooks, and figured I might as well convert the one I made for `macro_reduce.py` in a tutorial. I annotated the code a bit but could use some upgrade/elaboration at some point.

- **New features**:
    * New tutorial on macronization

### CODE REVIEW

**Comment by sduquemesa** [comment]
Great @soosub! 💫

If you could take it to a point where it can be showcased (maybe by adding an introductory paragraph explaining the context and objectives of the tutorial) I can add it to the tutorials gallery. What do you think?

File: doc/tutorials/tutorial_macro_reduce.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+RHG_macro.adj_generator(sparse=True)
+
+RHG_macro.draw()
+plt.show()
+
+
+######################################################################
+# As you can see, we replaced every node with a group of four nodes,

+# sweet!
```
Comment (sduquemesa) [inline doc/tutorials/tutorial_macro_reduce.py:77]: 🍭

File: doc/tutorials/tutorial_macro_reduce.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    
+    c = correct(code=RHG_code, decoder=decoder, weight_options=weight_options)
+    
+    successes += int(c)
+
+# Results
+error = (n_trials - successes) / n_trials
+print("Trials: ", n_trials)
+print("Successes: ", successes)
+print("Error rate: ", error)
```
Comment (sduquemesa) [inline doc/tutorials/tutorial_macro_reduce.py:117]: Just add a new line for formatting...
```suggestion
print("Error rate: ", error)

```

**Comment by soosub** [comment]
> 

Sounds good, I will convert it into a draft and open it once I'm happy with the content and formatting!

**Pull request closed without merging**